frontend/src/modules/dataset/dataset-browser.tsx

```diff
@@ -129,94 +125,70 @@ export function DatasetBrowser (props: DatasetBrowserProps) {
 
-    // Update the component if the datasetContext changes. This is expected to occur whenever the browser is going to display
-    // a different scope, dataset, or prefix.
-    useEffect(() => {
         const payload: Partial<DatasetBrowserState> = {
-            breadcrumbs: breadcrumbFromDataset(state.datasetContext, !!state.manageMode),
+            datasetContext: keepContext ? state.datasetContext : datasetContext,
             selectedItems: [],
-            isLoading: true
+            isLoading: false,
+            filteringText
         };
 
-        if (state.manageMode === DatasetBrowserManageMode.Create) {
-            payload.isLoading = false;
-        } else {
+        if (manageMode !== DatasetBrowserManageMode.Create) {
             switch (displayMode) {
                 case DatasetBrowserDisplayMode.Resource:
-                    // DatasetBrowser invariant requires state.datasetContext can never be undefined with this displayMode
-                    if (state.datasetContext) {
-                        fetchDatasetResources(state.datasetContext);
+                    if (shouldFetch && datasetContext) {
+                        payload.isLoading = true;
+                        if (keepItems) {
+                            fetchDatasetResources(datasetContext, state.nextToken, state.items);
+                        } else {
+                            fetchDatasetResources(datasetContext);
+                        }
                     }
                     break;
                 case DatasetBrowserDisplayMode.Dataset:
-                    fetchDatasets();
+                    if (shouldFetch && datasetContext.type) {
+                        payload.isLoading = true;
+                        fetchDatasets(datasetContext.type);
+                    }
                     break;
                 default:
                     payload.items = Object.keys(DatasetType).map((key) => ({ name: upperFirst(DatasetType[key]), type: DatasetType[key] }));
-                    payload.isLoading = false;
             }
         }
 
         setState({
             type: DatasetActionType.State,
             payload
         });
-    }, [state.datasetContext, state.manageMode, fetchDatasetResources, fetchDatasets, displayMode, state.refresh]);
-
-    const pageSize = 10;
-    // recalculate pages if filteredItems or pageSize changes
-    useEffect(() => {
-        setState({
-            type: DatasetActionType.Pagination,
-            payload: {
-                pagesCount: Math.max(1, Math.ceil(state.filter.filteredItems.length / pageSize)),
-            }
-        });
-    }, [state.filter.filteredItems, pageSize]);
+    }, [fetchDatasetResources, fetchDatasets, manageMode, state.datasetContext, state.filteringText, state.items, state.nextToken]);
 
-    // if pagination changes make sure currentPageIndex isn't greater than pageCount
+    const [firstLoad, setFirstLoad] = useState(true);
```

Comment:
Do we actually need this? If we only want to run `updateDatasetContext(state.datasetContext || {}, '', false);` once can we just pass an empty array to the `useEffect` instead?

Reply:
You absolutely can do that but you get an error from the linter because you are not listing things you're using. This is a little more verbose but I think the "correct" way to go about this. If your concern is just it is unnecessarily verbose I'm cool swapping that with just disabling the linter for that line.

Reply:
Yeah I think that's one lint rule that really shouldn't always apply so I'm generally fine if we add a comment as to why we're ignoring it. This is fine too so no big deal.